### ISSUE TITLE
Saved models dictionaries are incomplete  - No epoch indicator 

### ISSUE THREAD

**Issue by Camilochiang** [ultralytics/yolov5#6005]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Training

### Bug

When opening the models after training, several key are empty or wrong (as epoch = -1)
```
model_info = torch.load(PROJECT_PATH + '/exp16/weights/best.pt')
model_info['epoch'] 
-1
model_info['ema']
model_info['optimizer']
model_info['date']
'2021-12-12T22:06:54.098125'
## What indicate me that was epoch 76, in my case
## Repeating with the last one (trained till 300 epochs)
model_info = torch.load(PROJECT_PATH + '/exp16/weights/last.pt')
model_info['epoch']
-1
model_info['ema']
model_info['optimizer']
 model_info['date']
'2021-12-12T22:10:42.097877'
```

### Environment

YOLOv5 🚀 v6.0-127-g554f782 torch 1.10.0+cu102 CUDA:0 (Quadro RTX 5000 with Max-Q Design, 16125MiB)
Ubuntu 18.04
Python 3.6.9

### Additional

I tried to have a look why this could be happening but not sure why...

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@Camilochiang this is not a bug, I've removed the bug label. If these values were not removed upon training completion your checkpoint would be about 5X larger than it is, and you'd be raising another bug report about the excess size of your checkpoint.

See `strip_optimizer()` function for details.

https://github.com/ultralytics/yolov5/blob/c1249a47c7fe19e2067cb25ed8347e67d26ff1f1/utils/general.py#L737-L751

**Comment by glenn-jocher** [comment]
@Camilochiang YOLOv5 🚀 training checkpoints should be 4X the size of final trained checkpoints, as they carry not just a FP16 model, but a FP16 [EMA](https://www.tensorflow.org/api_docs/python/tf/train/ExponentialMovingAverage) and an FP32 optimizer of the same size as the model (each model parameter has its own FP32 gradient saved within the optimizer). Final checkpoints contain only an FP16 model, with the EMA and optimizer both stripped after the final epoch of training:
https://github.com/ultralytics/yolov5/blob/75feeb797c4a9553f4274860c6c278f1fc628f60/train.py#L435-L439

The `strip_optimizer()` function updates the checkpoint dictionary value for `model`, replacing it with `ema`, and sets `ema` and `optimizer` keys to `None`, which will reduce the checkpoint size by 3/4.
https://github.com/ultralytics/yolov5/blob/75feeb797c4a9553f4274860c6c278f1fc628f60/utils/general.py#L495-L509

Final trained checkpoints after `strip_optimizer()` should match the README table sizes below.
<p align="center"><img width="700" alt="YOLOv5 Models" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/model_comparison.png"></p>

Good luck 🍀 and let us know if you have any other questions!

**Comment by Camilochiang** [comment]
Oh, Great! ;)
